### DISCUSSION TITLE
Units of y-axis of integrate1d_ng()

### DISCUSSION THREAD

**Discussion by etortorici** [silx-kit/pyFAI#2161] [General]
Hello,

I am a relatively new pyFAI user, and I'm trying to understand, in detail, the units of the y-axis of the 1D integration of a WAXS image. I assumed it would be something like "counts per pixels contributing to bin" where a bin would be q or 2theta. When I program my own integration where I correct for the amount of pixels contributing to each bin, I get the right shape of data; however, the results I get are about 4x that of pyFAI's result.

I am wondering if there is a simple explanation of the algorithm and corrections used so that I can better understand the units of the y-axis of the output of the integration.

**Comment by kif** [comment]
Hi Teddy,

TLDNR: use `correctSolidAngle=False`

The intensity reported by `pyFAI` is by default the weighted average of the pixel within a ring. The weights are the product of the solid-angle, polarization factor, flatfield, absorption correction and  extra normalization factors. They are all set to 1 (neutral element for multiplication) EXCEPT for the solid-angle correction !
The solid-angle correction is indeed activated by default (can be disabled), the factor is exactly 1 at the PONI and slightly less elsewhere. It is also possible to use absolute solid-angle correction but the scaling factor is distance²/pixelsize² which makes the signal 10^10 large than the original image. you would have noticed.

Cheers

Jerome
